x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java

```diff
@@ -174,8 +174,12 @@ public Stream<Attribute> parallelStream() {
     }
 
     @Override
-    public boolean equals(Object o) {
-        return delegate.equals(o);
+    public boolean equals(Object obj) {
+        if (obj instanceof AttributeSet as) {
+            obj = as.delegate;
+        }
+
+        return delegate.equals(obj);
```

Comment:
Do we expect it to ever be equal if object is not `AttributeSet`? If not, following could avoid reassigning the parameter
```suggestion
        return obj instanceof AttributeSet as && Objects.equals(as.delegate, this.delegate);
```

Reply:
Yeah, I think your approach would be stricter and I'd write it like this if I was writing this from scratch.

However, that'd potentially rule previously equal sets to not be equal anymore, even though that probably never applied in reality. But we'd have to go and check. And this'd be inconsistent with `AttributeMap`, which also defers to `delegate.equals` even for instances of different classes.